prelude/cxx/tools/compilation_database.bxl

```diff
@@ -48,5 +60,17 @@ generate = bxl_main(
             cli_args.target_expr(),
             doc = "Targets to generate the database for",
         ),
+        "include-headers": cli_args.bool(
+            default = True,
+            doc = "Include header file compilation db actions"
+        ),
+        "add-target-name": cli_args.bool(
+            default = False,
+            doc = "Include the name of the target in the compilation database entries (non-standard)"
+        ),
+        "recursive": cli_args.bool(
+            default = False,
```

Comment:
This should probably default to `True`, or even not be a configuration option IMO.

I hadn't even considered that case because I always generate compile commands for the whole repo, but without this, a more selective target expression would lead to a compilation database where going to definition results in a broken experience.

For cases where tools only look at a single translation unit (e.g. clang-tidy), one should use the compile commands subtargets instead of this BXL script anyway.